jumpscale/sals/reservation_chatflow/deployer.py

```diff
@@ -1317,6 +1351,17 @@ def deploy_kubernetes_cluster(
 
         # deploy_master
         master_ip = ip_addresses[0]
+        # Reserve public_Ip on node_id[0]
+        public_id_wid = 0
+        master_public_ip = ""
+        if public_ip:
+            public_id_wid = self.get_public_ip(pool_id, node_ids[0], solution_uuid=metadata.get("solution_uuid"))
+            master_public_ip = (
```

Comment:
use the returned public ip from previous comment instead of fetching the workload info again